docs/compilers/CSharp/Compiler Breaking Changes - DotNet 6.md

```diff
@@ -0,0 +1,13 @@
+﻿## This document lists known breaking changes in Roslyn in C# 9.0 introduced with .NET 6.
+
+1. https://github.com/dotnet/roslyn/issues/46044 In C# 9.0 (Visual Studio 16.9), a warning is reported when assigning `default` to, or when casting a possibly `null` value to a type parameter type that is not constrained to value types or reference types. To avoid the warning, the type can be annotated with `?`.
```

Comment:
>In C# 9.0 [](start = 49, length = 9)

I think this warning is now produced even for LangVer=8, right? If so we should just say "In Visual Studio 16.9, ..."

Reply:
> I think this warning is now produced even for LangVer=8, right?

No, the warnings are not reported with `/langversion:8`.

> So we could start a new document for post-DotNet 5.

I've renamed the document to "Compiler Breaking Changes - post DotNet 5.md" based on our discussion.

---
In reply to: [572483433](https://github.com/dotnet/roslyn/pull/51070#discussion_r572483433) [](ancestors = 572483433)